### DISCUSSION TITLE
Model with custom root type fails if in a nested model

### DISCUSSION THREAD

**Discussion by NixBiks** [pydantic/pydantic#2480] [Question] [answered]
I'm not sure if this is a bug or not but it feels like a bug to me. Here is the example where the last line causes a `ValidationError` since `__root__` keyword is not specifically provided.

```python
from pydantic import BaseModel
from typing import Union, List


class A(BaseModel):
    name: str

class B(BaseModel):
    pipeline: List["Component"]

class Component(BaseModel):
    __root__: Union[A, B]

B.update_forward_refs(Component=Component)


Component.parse_obj({"name": "John Doe"})  # passes
Component.parse_obj({"pipeline": [{"__root__": {"name": "John Doe"}}]})  # passes

Component.parse_obj({"pipeline": [{"name": "John Doe"}]})  # fails
```

**Comment by NixBiks** [comment]
Actually now that I look at it, it sorta makes sense. I still wonder how to achieve my goal here in the best and most simple way.

**Comment by PrettyWood** [comment, marked as answer]
I thought I fixed this in `v1.8` 🤔 Which version do you use please?

**Reply by NixBiks** [reply]
I thought I was on `v1.8` but it was `v1.7`. It works as expected in `v1.8` - thanks!

Side note: apparently `poetry update pydantic@latest` does not work as expected
